### PR TITLE
Move config / Adjust flake8 settings

### PR DESCRIPTION
**Pull request by iamdefinitelyahuman**
### What I did
* move all config settings to `setup.cfg`
* add a `flake8` setting to ignore `F401` (imported but not used) on `__init__.py` modules - we are imporing as a way to expose the method "publicly" to other subpackages, so this warning is not important
* remove `# NOQA: F401` comments in `__init__.py` modules

### How to verify it
Look at the diff!

### Cute Animal Picture
image

### CODE REVIEW

**Comment by iamdefinitelyahuman** [comment]
One thing to note: i swapped spaces for tabs within `setup.cfg` - `bumpversion`  reaaaally likes tabs and will change it whenever called. I made the change now to lessen the diff later.

File: setup.cfg [hunk 1]
```diff
@@ -1,18 +1,56 @@
+[bumpversion]
+current_version = 0.1.0-beta.17
+commit = True
+tag = True
+parse = (?P<major>\d+)\.(?P<minor>\d+)\.(?P<patch>\d+)(-(?P<stage>[^.]*)\.(?P<devnum>\d+))?
+serialize =
```
Comment (fubuloubu) [inline setup.cfg:6]:
```suggestion
serialize = 
```

File: setup.cfg [hunk 2]
```diff
@@ -1,18 +1,56 @@
...
+	{major}.{minor}.{patch}-{stage}.{devnum}
+	{major}.{minor}.{patch}
+
+[bumpversion:part:stage]
+optional_value = stable
+first_value = stable
+values =
```
Comment (fubuloubu) [inline setup.cfg:13]:
```suggestion
values = 
```

File: setup.cfg [hunk 3]
```diff
@@ -1,18 +1,56 @@
...
+[bumpversion:file:setup.py]
+search = version='{current_version}',
+replace = version='{new_version}',
+
+[aliases]
+test = pytest
+
+[flake8]
+max-line-length = 100
+exclude =
```
Comment (fubuloubu) [inline setup.cfg:29]:
```suggestion
exclude = 
```

File: setup.cfg [hunk 4]
```diff
@@ -1,18 +1,56 @@
...
+	venv*
+	.tox
+	docs
+	build
+per-file-ignores =
```
Comment (fubuloubu) [inline setup.cfg:34]:
```suggestion
per-file-ignores = 
```

File: setup.cfg [hunk 5]
```diff
@@ -1,18 +1,56 @@
...
-          --hypothesis-show-statistics
+	--cov-branch
+	--cov-report term
+	--cov-report html
+	--cov-report xml
+	--cov=vyper
+	--hypothesis-show-statistics
 python_files = test_*.py
 testpaths = tests
 markers =
```
Comment (fubuloubu) [inline setup.cfg:55]:
```suggestion
markers = 
```

**Pull request merged**